src/game/objects/pickup.c

```diff
@@ -42,65 +66,156 @@ void SetupPickupObject(OBJECT_INFO *obj)
+                m_PickUpPosition.y = lara_item->pos.y - item->pos.y;
+                if (MoveLaraPosition(&m_PickUpPosition, item, lara_item)) {
+                    lara_item->anim_number = AA_PICKUP;
+                    lara_item->current_anim_state = AS_PICKUP;
+                    have_item = true;
+                }
+                g_Lara.interact_target.item_num = item_num;
+            } else if (
+                g_Lara.interact_target.is_moving
+                && g_Lara.interact_target.item_num == item_num) {
+                g_Lara.interact_target.is_moving = false;
+                g_Lara.gun_status = LGS_ARMLESS;
+            }
+            if (have_item) {
+                g_Lara.head_y_rot = 0;
+                g_Lara.head_x_rot = 0;
+                g_Lara.torso_y_rot = 0;
+                g_Lara.torso_x_rot = 0;
+                lara_item->frame_number =
+                    g_Anims[lara_item->anim_number].frame_base;
+                g_Lara.interact_target.is_moving = false;
+                g_Lara.gun_status = LGS_HANDSBUSY;
+            }
+        } else if (
+            g_Lara.interact_target.item_num == item_num
+            && lara_item->current_anim_state == AS_PICKUP) {
+            if (lara_item->frame_number == AF_PICKUP_ERASE) {
+                PickUp_GetItem(item_num, item, lara_item);
+            }
+        }
+    } else if (g_Lara.water_status == LWS_UNDERWATER) {
+        item->pos.x_rot = -25 * PHD_DEGREE;
+
+        if ((g_Input.action && lara_item->current_anim_state == AS_TREAD
+             && g_Lara.gun_status == LGS_ARMLESS
+             && !g_Lara.interact_target.is_moving)
+            || (g_Lara.interact_target.is_moving
+                && g_Lara.interact_target.item_num == item_num)) {
+
+            if (TestLaraPosition(m_PickUpBoundsUW, item, lara_item)) {
+                if (MoveLaraPosition(&m_PickUpPositionUW, item, lara_item)) {
+                    lara_item->anim_number = AA_PICKUP_UW;
+                    lara_item->current_anim_state = AS_PICKUP;
+
+                    lara_item->goal_anim_state = AS_TREAD;
+                    lara_item->frame_number =
+                        g_Anims[lara_item->anim_number].frame_base;
+                    g_Lara.interact_target.is_moving = false;
+                    g_Lara.gun_status = LGS_HANDSBUSY;
+                }
+                g_Lara.interact_target.item_num = item_num;
+            } else if (
+                g_Lara.interact_target.is_moving
+                && g_Lara.interact_target.item_num == item_num) {
+                g_Lara.interact_target.is_moving = false;
+                g_Lara.gun_status = LGS_ARMLESS;
+            }
+        } else if (
+            g_Lara.interact_target.item_num == item_num
+            && lara_item->current_anim_state == AS_PICKUP
+            && lara_item->frame_number == AF_PICKUP_UW) {
+            PickUp_GetItem(item_num, item, lara_item);
+            return;
```

Comment:
I don't think we need this `return;` considering how previous call to `PickUp_GetItem` in another branch continues with the restoration of the angles.

Reply:
I added the `return` because Lara was getting double items from underwater pickups. But, I had the original underwater pickup code and the new pickup code duplicated which I realized and removed last night. Good catch!